internal/reconcile/release.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcile
+
+import (
+	"errors"
+
+	helmrelease "helm.sh/helm/v3/pkg/release"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta2"
+	"github.com/fluxcd/helm-controller/internal/release"
+	"github.com/fluxcd/helm-controller/internal/storage"
+)
+
+var (
+	// ErrNoCurrent is returned when the HelmRelease has no current release
+	// but this is required by the ActionReconciler.
+	ErrNoCurrent = errors.New("no current release")
+	// ErrNoPrevious is returned when the HelmRelease has no previous release
+	// but this is required by the ActionReconciler.
+	ErrNoPrevious = errors.New("no previous release")
+	// ErrReleaseMismatch is returned when the resulting release after running
+	// an action does not match the expected current and/or previous release.
+	// This can happen for actions where targeting a release by version is not
+	// possible, for example while running tests.
+	ErrReleaseMismatch = errors.New("release mismatch")
+)
+
+// observeRelease returns a storage.ObserveFunc which updates the Status.Current
+// and Status.Previous fields of the HelmRelease object. It can be used to
+// record Helm install and upgrade actions as - and while - they are written to
+// the Helm storage.
+func observeRelease(obj *helmv2.HelmRelease) storage.ObserveFunc {
```

Comment:
the naming is similar to ObserveRelease. Maybe a better naming for this function that returns an `ObserveFunc` would be `observeReleaseFunc`?

Reply:
Why? `ObserveRelease` comes from another package so is always prefixed. None of the other functions are suffixed with `func`.